reana_server/reana_admin.py

```diff
@@ -464,3 +471,96 @@ def list_quota_usage(
+    """List quota resources."""
+    resource_names = [resource.name for resource in Resource.query]
+    click.echo("\n".join(resource_names))
+
+
+@reana_admin.command("quota-set", help="List available quota resources.")
+@click.option("--id", "id_", help="The id of the user.")
+@click.option("-e", "--email", help="The email of the user.")
+@click.option("--resource-name", "-r", help="Name of resource.", required=True)
+@click.option(
+    "--limit", "-l", help="New limit in canonical unit.", required=True, type=int
+)
+@admin_access_token_option
+@click.pass_context
+def set_quota_limit(ctx, admin_access_token, id_, email, resource_name, limit):
+    """Set quota limit per resource."""
+    try:
+        error_msg = None
+        user = _get_user_by_criteria(id_, email)
+        resource = Resource.query.filter_by(name=resource_name).one_or_none()
+        if not user:
+            error_msg = f"ERROR: Provided user {id_ or email} does not exist."
+        elif not resource:
+            error_msg = (
+                "ERROR: Provided resource name does not exist. Available "
+                f"resources are {[resource.name for resource in Resource.query]}."
+            )
+        if error_msg:
+            click.secho(
+                error_msg, fg="red", err=True,
+            )
+            sys.exit(1)
+
+        user_resource = UserResource.query.filter_by(
+            user=user, resource=resource
+        ).one_or_none()
+        if user_resource:
+            user_resource.quota_limit = limit
+            Session.add(user_resource)
+        else:
+            # Create user resource in case there isn't one. Useful for old users.
+            user.resources.append(
+                UserResource(
+                    user_id=user.id_,
+                    resource_id=resource.id_,
+                    quota_limit=limit,
+                    quota_used=0,
+                )
+            )
+        Session.commit()
+        click.secho(
+            f"Quota limit for {resource.name} successfully set for user {user.id_} ({user.email}).",
+            fg="green",
+        )
+    except Exception as e:
+        logging.debug(traceback.format_exc())
+        logging.debug(str(e))
+        click.echo(
+            click.style("Quota could not be set: \n{}".format(str(e)), fg="red"),
+            err=True,
+        )
+
+
+@reana_admin.command(
+    "quota-set-default-limits",
+    help="Set default quota limits to users that do not have any.",
+)
+@admin_access_token_option
+@click.pass_context
+def set_default_quota_limit(ctx, admin_access_token):
+    """Set default quota limits to users that do not have any."""
+    output_commands = []
+    users_without_quota_limits = User.query.filter(~User.resources.any()).all()
+    for user in users_without_quota_limits:
+        for resource in Resource.query:
+            output_commands.append(
+                f"flask reana-admin quota-set -e {user.email} -r '{resource.name}' "
```

Comment:
This turns out to be very slow as Python+Click has to be initialised for every user. Also, this would be very hard on the DB as each command would be instantiating a new DB session.

A solution for this could be to make `flask reana-admin quota-set` accept a list of ids/emails and perform the quota set by invoking the command from Python passing `users_without_quota_limits ` to it. Another option is to share part of the implementation with `quota-set` (extract to a function) and do everything in pure Python/SQLAlchemy and forget about invoking with Click.